website/blog/2021-02-22-7.13.0.md

```diff
@@ -0,0 +1,255 @@
+
+However, `loose` has some problems that lead to user frustration/confusion: the term itself does nothing to describe how it affects the compiler's behavior, and worse, sometimes it must be set the same across different plugins to ensure everything compiles.
+
+To help fix these problems, we've added a new top level option: `assumptions`! Similar to the new `targets` option, every plugin will now receive the assumptions you've enabled, eliminating the need to set the option individually. This is really valuable since a plugin can be influenced by multiple assumptions, and an assumption can influence multiple plugins.
+
+> :warning: This is **advanced** functionality. Please be careful when enabling assumptions, because they are _not_ spec-compliant and may break your code in unexpected ways.
+
+For example, when transforming optional chaining and nullish coalescing, Babel will generate this output by default:
+
+<table>
+<tr>
+<th>Input</th>
+<th>Output</th>
+</tr>
+<tr>
+<td>
+
+```js
+let user, points;
+
+let score = points ?? 0;
+let name = user?.name;
+```
+
+</td>
+<td>
+
+```js
+let user, points;
+
+let score = points !== null && points !== void 0 ? points : 0;
+let name = user === null || user === void 0 ? void 0 : user.name;
+```
+
+</td>
+</tr>
+</table>
+
+However, enabling the `noDocumentAll` assumption tells Babel "I'm never using `document.all`, so you can compile without worrying about it":
+
+```json
+{
+  "targets": "firefox 70",
+  "assumptions": { "noDocumentAll": true },
+  "presets": ["@babel/preset-env"]
+}
+```
+
+<table>
+<tr>
+<th>Input</th>
+<th>Output</th>
+</tr>
+<tr>
+<td>
+
+```js
+let user, points;
+
+let score = points ?? 0;
+let name = user?.name;
+```
+
+</td>
+<td>
+
+```js
+let user, points;
+
+let score = points != null ? points : 0;
+let name = user == null ? void 0 : user.name;
+```
+
+</td>
+</tr>
+</table>
+
+Check out the full list of assumptions [over in our documentation](https://babeljs.io/assumptions), where you can individually enable or disable them to see how they affect the compiled output.
+
+### Records and Tuples support ([#12145](https://github.com/babel/babel/pull/12145))
+
+Thanks to a collaboration with [Bloomberg](https://www.techatbloomberg.com/), Babel now supports transforming the "Records and Tuples" stage 2 [proposal](https://github.com/tc39/proposal-record-tuple).
+
+The Babel plugin transforms the records and tuples syntax using the global `Record` and `Tuple` functions:
+
+<table>
+<tr>
+<th>Input</th>
+<th>Output</th>
+</tr>
+<tr>
+<td>
+
+```js
+let data = #{
+  name: "Babel",
+  ids: #[1, 2, 3]
+};
+```
+
+</td>
+<td>
+
+```js
+let data = Record({
+  name: "Babel",
+  ids: Tuple(1, 2, 3)
+});
+```
+
+</td>
+</tr>
+</table>
+
+This means that you will need to load a polyfill for those global functions, such as [`@bloomberg/record-tuple-polyfill`](https://www.npmjs.com/package/@bloomberg/record-tuple-polyfill).
```

Comment:
Some additional context/links to documentation around using 3rd party polyfills might be useful here. As a reader, some questions I might have are:

1. What does "load a polyfill" mean?
1. How do I use Babel with a 3rd party polyfill?
1. Why isn't this covered by `targets`?

Reply:
```suggestion
This means that you will need to load a polyfill for those global functions, such as [`@bloomberg/record-tuple-polyfill`](https://www.npmjs.com/package/@bloomberg/record-tuple-polyfill), either by importing it in your code or by adding it as an entrypoint to your bundler's configuration.

> **NOTE**: No engine currently supports records and tuples, so you always need to load the polyfill.
```

Does this make it clearer?

Reply:
Yes!